lib/serverless/aws-lambda.js

```diff
@@ -194,16 +192,20 @@ class AwsLambda {
         throw err
       }
       if (shim.isPromise(res)) {
-        res = lambdaInterceptPromise(res, txnEnder)
+        res = lambdaInterceptPromise(res, resultProcessor, txnEnder)
       }
       return res
     }
 
     // In order to capture error events
     // we need to store the error in uncaughtException
     // otherwise the transaction will end before they are captured
-    function lambdaInterceptPromise(prom, cb) {
+    function lambdaInterceptPromise(prom, resultProcessor, cb) {
       return prom.then(function onThen(arg) {
+        // check if we are in web transaction
```

Comment:
Seems this should only occur for an uncaught exception, so I believe it is fine to only process the result in the 'then'.

Reply:
OK, do you mean I can remove the comment here?

Reply:
I was just commenting out loud that I *think* this is fine, IF the catch is only called on uncaught exceptions like I think it is.

That being said, I don't think the comment is necessary and could potentially be confusing in the future if we added a result processor for non-web case for some reason. (Really though, this instrumentation could prob use some restructuring at some point now that it has evolved a few times). So I'd be fine if you removed the comment.